### DISCUSSION TITLE
Fundamental Changes in Jamku v4

### DISCUSSION THREAD

**Discussion by adarshmadrecha** [madrecha/jamku#864] [Announcements]
We have received lot of questions on what changes coming to Jamku v4 (beta release in Q2 2022). This post serves as live document summarising the changes being done in v4 as compared to v3

1. _Email/Google/Mobile Authentication_
Currently only mobile based authentication is supported. v4 will add ability to use your google account or email-n-password based authentication apart from mobile-n-password. 
Authentication will be moved to [Firebase](https://firebase.google.com/docs/auth), a Google Service for enhanced security and better protection against [DDoS attack](https://www.cloudflare.com/en-in/learning/ddos/what-is-a-ddos-attack/).

2. _Client Group will be Optional_
Currently even for Single Client, we need to create Client Group. Client Grouping will become optional. So, users onboarding into Jamku, will have faster onboarding experience. Grouping only needs to be done as per requirement.

3. _Billing will be moved to Client Level_
Since Client Group is now optional, this is a required change we need to do. Client Location will now compulsorily be defined at Client Level.
`Impact`: Billing for more than one client in a single bill won't be possible.

4. _Separate UI for Mobile and Desktop_
Currently, the UI for Mobile and Desktop is remarkably similar. While it makes learning mobile app super easy, the experience is not perfect. We will now be developing Mobile app separately from the Desktop App.

**Comment by vatsalbhatia** [comment]
Point No. 3 is recomended but do not part away with the facility to Bill more than one client in a single bill, this is equally important however bill at client level needed too.
